### ISSUE TITLE
isCurrentMonth and isCurrentQuarter ambiguity and inconsistency

### ISSUE THREAD

**Issue by Sarke** [briannesbitt/Carbon#1314]
I've noticed that the *some* `isCurrent` methods do not act like one would assume.  

The following do work as expected:

```
isCurrentYear
isCurrentDay
isCurrentHour
isCurrentMinute
isCurrentSecond
```

However, the following is an example of what I would call unexpected results:

```php
$dt = new Carbon\Carbon('1995-05-22');
var_dump($dt->isCurrentMonth()); // true
```

The affected methods are:

```
isCurrentMonth
isCurrentQuarter
```

Part of this was covered here:

https://github.com/briannesbitt/Carbon/issues/1248#issuecomment-382138876
https://github.com/briannesbitt/Carbon/issues/1248#issuecomment-382662788

And I can understand the argument that `isSameMonth` *can* mean "same month any year".  However, I do not believe this argument applies when saying "current month", which to me *only* applies to one specific month (like the other `isCurrent` methods above).

A quick search for `isCurrentMonth` on github shows that other implementations agree with me that the year should match as well.

Personally I would like to see the `$ofSameYear` parameter (for `isSameMonth` and `isSameQuarter`) change it's default value to `true`.  Do it, it's the right thing! :p

Doing that could have some backwards compatibility issues.  Similar but less serious would be if we change the `isCurrentMonth` implementation to use `return $this->isSameMonth(null, true);` which I also think would make perfect sense and is IMO by far the best solution to fix this.

Frankly I can't really see many people using `isCurrentMonth` when they are really trying to match the same month any year; they would have used `isSameMonth` for that.  I think the BC issues would be minimal.

However, if you don't agree about the BC impact, another alternative would be to implement an option, similar to `useMonthsOverflow`, that would set the default value of the `$ofSameYear` parameter.  This would be more of a workaround though, since the inconsistency and counterintuitiveness of `isCurrentMonth` and `isCurrentQuarter` would still be there.

The only way to truly correct everything, including ambiguity, *and* cover any BC issues would be to deprecate those two methods and replace them with something like `isCurrentMonthOnly` and `isCurrentMonthAndYear`.

Similarly `isSameMonthOnly` and `isSameMonthAndYear`, because the `isSameMonth` is not consistent with the `isSameYear` and `isSameDay` behaviours.

Thanks.

**Comment by kylekatarnls** [comment]
The default value will be ˋtrueˋ in Carbon 2. It's planned. Our BC policy is to keep existing unit tests pass until next major release. So the method to set the default value could be our best option for version 1.

**Comment by kylekatarnls** [comment]
Will be fixed in Carbon 1 via #1323:
Run `Carbon::compareYearWithMonth();` to compare month with year by default.
Will be fixed in Carbon 2 via #1315
Mont-year comparison is the default behavior.

**Comment by Sarke** [comment]
Awesome, thank you!
